tiled/src/main/kotlin/ktx/tiled/batchTiledMapRenderer.kt

```diff
@@ -0,0 +1,74 @@
+package ktx.tiled
+
+import com.badlogic.gdx.graphics.OrthographicCamera
+import com.badlogic.gdx.maps.tiled.renderers.BatchTiledMapRenderer
+import com.badlogic.gdx.maps.tiled.tiles.AnimatedTiledMapTile
+import com.badlogic.gdx.math.Matrix4
+import com.badlogic.gdx.math.Rectangle
+
+/**
+ * Automatically calls [BatchTiledMapRenderer.beginRender] and [BatchTiledMapRenderer.endRender].
+ * @param camera A camera to set on the renderer before [BatchTiledMapRenderer.beginRender].
+ * @param block inlined. Executed after [BatchTiledMapRenderer.beginRender] and before [BatchTiledMapRenderer.endRender].
+ */
+inline fun <T : BatchTiledMapRenderer> T.use(
+  camera: OrthographicCamera,
+  block: (T) -> Unit
+) {
+  this.setView(camera)
+  AnimatedTiledMapTile.updateAnimationBaseTime()
```

Comment:
I'm wondering whether this update should be locked under a boolean flag such as `updateAnimation`. Could default to `true` if this is always expected to be called before using the renderer.

Reply:
It is called per default in LibGDX and from my understanding it is linked to the animated tiles that you can create in Tiled map editor. I personally find it very weird that this is included inside a `beginRender` method but I guess they expected that people will just call `render` most of the time which renders the entire map and includes one update of the animation timer, which does make sense.

However, imo, most games will split between background and foreground layer rendering which means you need to take care of `batch.begin`, `batch.end` yourself and at least I fell into the trap once that I forgot to call this static AnimatedTileMapTile update method :D

Anyway, to summarize, imo it does not hurt to have it as a parameter with default value true. I will include it :)

Reply:
edit: nevermind; if we make the cheesy solution of yours then I anyway cannot control anymore if it gets called or not because it is part of `beginRender` of the map renderer implementation ;)